tchannel/peer_heap.py

```diff
@@ -67,7 +84,28 @@ def pop_peer(self):
         return heap.pop(self)
 
     def push_peer(self, peer):
-        """Push a new peer into the heap"""
+        """Push a new peer into the heap
+
+        Order is equal to peer heap's current order plus random value within
+        the current peer size. It solves two problems when peers are in
+        same score:
+
+        1. 'dead peer' caused by random problem. If use fully random value, and
+        one peer is very unlock to get the worst value, as the peer selection
+        keeps working, that peer may not be selected anymore.
+
+        2. deterministic problem. If we just use push order, then all the
+        tchannel instance will follow the same pattern. If they all get the
+        same list of hyperbahn nodes, the peers that they pick to talk to
+        hyperbahn will be same. In that situation, it will cause the hyperbahn
+        node received very imbalance requests from tchannel clients.
+
+        All in all, it will keep certain level randomization but at the same
+        time make the peer score not deterministic among different tchannel
+        instance. For
+        """
+        self.order += 1
+        peer.order = self.order + random.randint(0, self.size())
```

Comment:
I'm confused. How does this solve problem 2? `peer.order` doesn't track the exact insertion order but it's still monotonically increasing with respect to peer insertion. So if we `push(A)`, and then `push(B)`, we will always have `A.order < B.order`, which means that score collisions will always resolve in a deterministic way?